app/com/linkedin/drelephant/exceptions/AnalyzedLog.java

```diff
@@ -0,0 +1,172 @@
+                        this.state = "FAILED";
+                        m = match("Task failed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
+                }
+
+                else if (match("Job failed as tasks killed").find()){                                                   // Killed MR Job log
+                        this.state = "KILLED";
+                        m = match("Task killed (task_[0-9]+_[0-9]+_[mr]_[0-9]+)");
+                        while(m.find()){
+                                allMatches.add(m.group(1));
+                        }
+                        this.list = allMatches;
+                        this.exception.setExceptionChain(matchExceptionChain(".*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*"));
+
+                }
+
+                else {                                                                                                  // Failed MR Task Task log
+                        this.state="FAILED";
+                        this.exception.setExceptionChain(matchExceptionChain("Error: (.*\\n(?:.*\\tat.+\\n)+(?:.*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*)*)"));
+                }
+
+        }
+
+
+        private List<List<String>> matchExceptionChain (String pattern){
+                Matcher m = Pattern.compile(pattern).matcher(log);
+                List<List<String>> exceptionChain = new ArrayList<List<String>>();
+                if(m.find()){
+
+                        for (String exceptionString : stringToExceptionChain(m.group())) {
+                                exceptionChain.add(stringToExceptionEvent(exceptionString));
+                        }
+                        return exceptionChain;
+                }
+                logger.info(exceptionChain);
+                return null;
+        }
+
+
+        public Matcher match(String pattern){
+                Matcher m;
+                return m = Pattern.compile(pattern).matcher(log);
+        }
+
+        public List<String> stringToExceptionChain (String s){
+               List<String> chain = new ArrayList<String>();
+               Matcher matcher = Pattern.compile(".*^(?!Caused by).+\\n(?:.*\\tat.+\\n)+").matcher(s);
+                while(matcher.find()){
+                        chain.add(matcher.group());
+                }
+                matcher = Pattern.compile(".*Caused by.+\\n(?:.*\\n)?(?:.*\\s+at.+\\n)*").matcher(s);
+                while(matcher.find()){
+                        chain.add(matcher.group());
+                }
+                return chain;
+        }
+
+        public List<String> stringToExceptionEvent(String s){
```

Comment:
whats the difference between ExceptionEvent and an ExceptionChain ? Add comments to explain.